tests/platform_tests/cli/test_show_platform_dpu.py

```diff
@@ -0,0 +1,183 @@
+    # Checking the state of dpu health again after bringing up interface and waiting for db entry to get populated
+    dpu_db_after = wait_until(60, 60, 0, count_dpu_modules_in_system_health_cli, duthost)
+
+    pytest_assert(dpu_db_before == count_up == count_down == dpu_db_after, "Link Flap is Tested'{}'".format(duthost.hostname))
+
+def test_shutdown_power_up_dpu(duthosts, enum_rand_one_per_hwsku_hostname, platform_api_conn):
+    """
+    @summary: Verify `shut down and power up DPU`
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    num_modules = int(chassis.get_num_modules(platform_api_conn))
+
+    for index in range(num_modules):
+        dpu = module.get_name(platform_api_conn, index)
+        duthosts.shell("config chassis modules shutdown %s"%dpu)
+        time.sleep(2)
+
+    pytest_assert(wait_until(120, 60, 0, check_dpu_module_status, duthost, num_modules, "off"),
+                          "Not all DPUs operationally down")
+
+    for index in range(num_modules):
+        dpu = module.get_name(platform_api_conn, index)
+        duthosts.shell("config chassis modules startup %s"%dpu)
+        time.sleep(2)
+
+    pytest_assert(wait_until(120, 60, 0, check_dpu_module_status, duthost, num_modules, "on"),
+                          "Not all DPUs operationally up")
+
+def test_reboot_cause(duthosts, enum_rand_one_per_hwsku_hostname, platform_api_conn):
+    """
+    @summary: Verify `Reboot Cause`
+    """
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+    num_modules = int(chassis.get_num_modules(platform_api_conn))
+
+    for index in range(num_modules):
+        dpu_name = module.get_name(platform_api_conn, index)
+        output_shutdown_dpu = duthost.shell("config chassis module shutdown %s"%(dpu_name))["stdout_lines"]
+        time.sleep(2)
+        output_startup_dpu = duthost.shell("config chassis module startup %s"%(dpus_name))["stdout_lines"]
+
+    pytest_assert(wait_until(120, 60, 0, check_dpu_reboot_cause, duthost, num_modules),
+                          "Not all DPUs operationally up")
+
+def test_pcie_link(duthosts, enum_rand_one_per_hwsku_hostname):
+    """
+    @summary: Verify `PCIe link`
+    """
+    CMD_PCIE_INFO = "show platform pcieinfo -c"
+
+    duthost = duthosts[enum_rand_one_per_hwsku_hostname]
+
+    logging.info("Verifying output of '{}' on '{}'...".format(CMD_PCIE_INFO, duthost.hostname))
+    output_pcie_info = duthost.command(CMD_PCIE_INFO)["stdout_lines"]
+    pytest_assert("PASSED" == output_pcie_info[-1], "PCIe Link is good'{}'".format(duthost.hostname))
+
+    num_modules = int(chassis.get_num_modules(platform_api_conn))
+
+    for index in range(num_modules):
+        dpu = module.get_name(platform_api_conn, index)
+        duthosts.shell("config chassis modules shutdown %s"%dpu)
+        time.sleep(2)
+
+    pytest_assert(wait_until(120, 60, 0, check_dpu_module_status, duthost, num_modules, "off"),
+                          "Not all DPUs operationally down")
+
```

Comment:
Should here be a check for the pcie after the DPUs are shutdown? 

This check is in the testplan:
### 1.7 Check removal of pcie link between NPU and DPU

#### Steps
 * Use `show platform pcieinfo -c` to run the pcie info test to check everything is passing
 * Use command `config chassis modules shutdown DPU<DPU_NUM>` to bring down the dpu (This will bring down the pcie link between npu and dpu)
 * Use `show platform pcieinfo -c` to run the pcie info test to check pcie link has been removed
 * Use command `config chassis modules startup DPU<DPU_NUM>` to bring up the dpu (This will rescan pcie links)
 * Use `show platform pcieinfo -c` to run the pcie info test to check everything is passing
 * This test is to check the PCie hot plug functinality since there is no OIR possible

Reply:
Added the check according to test plan.